### PR TITLE
chore(manager/terraform): use of-the-shelf parser for HCL parsing

### PR DESCRIPTION
**Pull request by secustor**
## Changes:
This PR moves the terraform manager away from custom regex based parsing to a proper of the shelf parser. 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -169,12 +169,14 @@
...
     "handlebars": "4.7.7",
     "hasha": "5.2.2",
+    "hcl2-parser": "1.0.3",
     "ignore": "5.2.0",
     "ini": "2.0.0",
     "js-yaml": "4.1.0",
     "json-dup-key-validator": "1.0.3",
     "json-stringify-pretty-compact": "3.0.0",
     "json5": "2.2.0",
+    "lodash": "4.17.21",
```
Comment (viceice) [inline package.json:179]: Why do we need this? 😕

Comment (secustor) [reply package.json:179]: This is a leftover from my prototyping. I'm going to remove it.

**Review by rarkins — COMMENTED** [review]
Once you're satisfied with this PR, would it be possible to split the test changes into a separate PR? The ideal would be:
- First PR which changes only tests, no change to code
- Second PR which changes only code, no significant changes to tests

**Comment by secustor** [comment]
> Once you're satisfied with this PR, would it be possible to split the test changes into a separate PR? The ideal would be:

I have not looked deeper in the changes, but nearly all the changes seem to come from order changes in the dependencies.    
Would it fine for you to sort the output of the existing tests?

**Review by viceice — COMMENTED** [review]
We are ok with snapshot order change here, but we should extend tests in separate PR to validate deps count stays same. we should also add some partial snapshots matches to verify ( jest will ignore property order i think )

**Comment by secustor** [comment]
> We are ok with snapshot order change here, but we should extend tests in separate PR to validate deps count stays same. 

The fact that the number of total deps is lower is expected  in the single test, where I have changed.        
Test: https://github.com/renovatebot/renovate/pull/14262/files#diff-c432fd3c491899e82e6d47518ff5c3922c2fb3d37726564c1acdd0ee976b9708R78
Fixture: https://github.com/renovatebot/renovate/blob/main/lib/manager/terraform/__fixtures__/docker.tf#L50

This is the case as the library does not return `not_supported_resource` as a resource anymore.  

> we should also add some partial snapshots matches to verify ( jest will ignore property order i think )

I have tested this locally. Sadly Jest fails if the order of elements are not correct, even if the snapshot is supplied as parameter.

**Comment by viceice** [comment]
So we can at least extract the `spec.ts` changes to a separate PR? So can can validate to find the same valid deps?

**Comment by rarkins** [comment]
Closing due to age and conflicts, and trying to lower open PRs in the repo. Welcome to reopen/recreate of course

**Pull request closed without merging**